shell-ui/src/auth/AuthProvider.js

```diff
@@ -39,6 +57,9 @@ function OAuth2AuthProvider({ children }: { children: Node }) {
     loadUserInfo: true,
     automaticSilentRenew: true,
     monitorSession: false,
+    MetadataServiceCtor: authConfig.defaultDexConnector
+      ? defaultDexConnectorMetadataService(authConfig.defaultDexConnector)
```

Comment:
It's more a question: 
If we want to use Dex to authenticate the users against a connector-based interface, how should we set the `defaultDexConnector` in `runtime-app-configuration`?

Reply:
Didn't we start documenting all this rework of shell-ui? Would be useful IMO, both now for reviewers catching up with the progress, and in the future for maintenance / onboarding.

Reply:
`defaultDexConnector` should take the connector id as defined in dex configuration.
For example given the following Dex connectors configuration :
```
       connectors:
            - type: oidc
              id: keycloak
              name: Keycloak
              config:
                   //....
```

Providing `keycloak` as `defaultDexConnector` in `runtime-app-configuration` will automatically redirect the user to this connector.

Reply:
I've opened #3476 regarding the documentation and will address it as soon as possible